File: jmlparser-jml-tools/src/main/java/com/github/jmlparser/lint/LintRule.java
package com.github.jmlparser.lint;

import com.github.javaparser.ast.Node;

/**
 * @author Alexander Weigl
 * @version 1 (12/29/21)
 */
public interface LintRule {
    String HINT = "HINT";
    String WARN = "WARN";
    String ERROR = "ERROR";

    void accept(Node node, LintProblemReporter problemReporter);
}

File: tools/lint/src/main/kotlin/io/github/jmltoolkit/lint/LintRule.kt
package io.github.jmltoolkit.lint

import com.github.javaparser.ast.Node

/**
 * @author Alexander Weigl
 * @version 1 (12/29/21)
 */
interface LintRule {
    fun accept(node: Node, problemReporter: LintProblemReporter, config: JmlLintingConfig)

    companion object {
        const val HINT: String = "HINT"
        const val WARN: String = "WARN"
        const val ERROR: String = "ERROR"
    }
}
